docker-entrypoint.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env sh
+set -eu
+
+
+# Set correct HTML base tag, so static resources are fetched
+# from the right path instead of the root path.
+# NOTE: Trailing and leading slashes in base href are important!
+# Using `~` separator to avoid problems with forward slashes
+
+if [-z "$BASE_URI"]; then
+BASE_URI="/"
+fi 
+if [ $BASE_URI != "/" ]; then 
+    mkdir -p /usr/share/nginx/$BASE_URI
+    mv /usr/share/nginx/html/* /usr/share/nginx/$BASE_URI/ 
+    mv  /usr/share/nginx/$BASE_URI /usr/share/nginx/html/
+    sed --in-place 's~<base href="/">~<base href="/'${BASE_URI}'/">~' /usr/share/nginx/html/$BASE_URI/index.html
+fi
+exec "$@"
```

Comment:
Does this chain to `nginx -g  "demon: off;"` or to the original entry point. We need the latter to handle the `nginx.d/*.template` line above (otherwise the nginx config will be invalid).

Reply:
> Does this chain to `nginx -g "demon: off;"` or to the original entry point. We need the latter to handle the `nginx.d/*.template` line above (otherwise the nginx config will be invalid).

 I have tested it on local machine and config files are intact and it doesnt change. It first hits the entry point and then go to cmd. Please find the snapshot of the local running instance.


Environment
BASE_URI
/

PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin

NGINX_VERSION
1.22.1

PKG_RELEASE
1

NJS_VERSION
0.7.11

server_name
localhost

bkg_backend
http://127.0.0.1:9090/bkg/v2

ebl_backend
http://127.0.0.1:9090/ebl/v3

Reply:
It is a problem if the `.template` is **not** substituted. It is not clear to me if it is or not.

Please provide a CI/github action test case that shows the image can be run and curl can hit the index.html (getting a HTTP 200).

Reply:
> Also, I am starting to think we should have a CI test for this. A simple `docker run ...` followed by a `curl -I http://localhost/<chosen-base-path>` and some assertion that the result it a HTTP 200 OK

Sure.. i will add one CI test to check if its working as expected.